src/main/java/com/celements/common/observation/listener/LocalEventListener.java

```diff
@@ -15,26 +20,65 @@
 @Component(LocalEventListener.NAME)
 public class LocalEventListener extends org.xwiki.observation.remote.internal.LocalEventListener {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(LocalEventListener.class);
+  final Logger LOGGER;
 
   public static final String NAME = "observation.remote";
 
+  final Map<Class<? extends Event>, Long> lastLoggedMap = new ConcurrentHashMap<>();
+
+  public LocalEventListener() {
+    LOGGER = LoggerFactory.getLogger(LocalEventListener.class);
+  }
+
+  /**
+   * for test purposes only
+   */
+  LocalEventListener(Logger logger) {
+    LOGGER = logger;
+  }
+
   @Override
   public String getName() {
     return NAME;
   }
 
   @Override
   public void onEvent(Event event, Object source, Object data) {
-    if (event.getClass().isAnnotationPresent(Local.class)) {
+    Class<? extends Event> type = event.getClass();
+    if (type.isAnnotationPresent(Local.class)) {
       LOGGER.info("skipping local event '{}'", event.getClass());
     } else {
-      try {
-        super.onEvent(event, source, data);
-      } catch (Exception exc) {
-        LOGGER.error("failed to notify RemoteObservationManager", exc);
+      if (checkSerializability(type, source, data)) {
+        try {
+          super.onEvent(event, source, data);
+        } catch (Exception exc) {
+          LOGGER.error("unable to notify remote event [{}], source [{}], data [{}]", type, source,
+              data, exc);
+        }
       }
     }
   }
 
+  boolean checkSerializability(Class<? extends Event> type, Object source, Object data) {
+    boolean serializable = true;
+    String msg = "unable to notify remote event [{0}]";
+    if (!(source instanceof Serializable)) {
+      msg += ", source [{1}] not serializable";
+      serializable = false;
+    }
+    if (!(data instanceof Serializable)) {
```

Comment:
"instanceof Serializable" is not sufficient for checkSerializability. Please test by serializing the data and catching the exception.